From 41c1d9a7e002990123e05c081986237197d29fc5 Mon Sep 17 00:00:00 2001
From: Huang Chenyi <b44402@freescale.com>
Date: Fri, 17 May 2013 13:46:48 +0800
Subject: [PATCH 199/310] ENGR00262859 Reserve 32K space for data partition
 after format file system

Reserve 32K space for data partiton in order to
support data partition encryption

Signed-off-by: Huang Chenyi <b44402@freescale.com>
---
 common/tools/fsl-sdcard-partition.sh | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/common/tools/fsl-sdcard-partition.sh b/common/tools/fsl-sdcard-partition.sh
index ff8185c..2012acb 100755
--- a/common/tools/fsl-sdcard-partition.sh
+++ b/common/tools/fsl-sdcard-partition.sh
@@ -87,6 +87,18 @@ function format_android
     mkfs.ext4 ${node}5 -Lsystem
     mkfs.ext4 ${node}6 -Lcache
     mkfs.ext4 ${node}7 -Lvender
+    mkdir /media/tmp
+    mount ${node}4 /media/tmp
+    amount=$(df -k | grep ${node}4 | awk '{print $2}')
+    stag=$amount
+    stag=$((stag-32))
+    kilo=K
+    amountkilo=$stag$kilo
+    sleep 1s
+    umount /media/tmp
+    rm -rf /media/tmp
+    e2fsck -f ${node}4
+    resize2fs ${node}4 $amountkilo
 }
 
 function flash_android
-- 
1.8.0

